docs/06-ReusingTestCode.md

```diff
@@ -0,0 +1,636 @@
+
+Codeception can generate a PageObject class for you with command:
+
+```bash
+$ php codecept.phar generate:pageobject Login
+```
+
+This will create a `Login` class in `tests/_support/Page`. The basic PageObject is nothing more then empty class with a few stubs.
+It is expected you will get it populated with UI locators of a page it represents and then those locators will be used on a page.
+Locators are represented with public static properties:
+
+```php
+<?php
+namespace Page;
+
+class Login
+{
+    public static $URL = '/login';
+
+    public static $usernameField = '#mainForm #username';
+    public static $passwordField = '#mainForm input[name=password]';
+    public static $loginButton = '#mainForm input[type=submit]';
+}
+?>
+```
+
+And this is how this page object can be used in a test:
+
+```php
+<?php
+use Page\Login as LoginPage;
+
+$I = new AcceptanceTester($scenario);
+$I->wantTo('login to site');
+$I->amOnPage(LoginPage::$URL);
+$I->fillField(LoginPage::$usernameField, 'bill evans');
+$I->fillField(LoginPage::$passwordField, 'debby');
+$I->click(LoginPage::$loginButton);
+$I->see('Welcome, bill');
+?>
+```
+As you see, you can freely change markup of your login page, and all the tests interacting with this page will have their locators updated according to properties of LoginPage class.
+
+But let's move further. A PageObject concept also defines that methods for the page interaction should also be stored in a PageObject class. It now stores passed instance of Actor class. An AcceptanceTester can be accessed via `AcceptanceTester` property of that class. Let's define a `login` method in this class.
+
+```php
+<?php
+class UserLoginPage
+{
+    // include url of current page
+    public static $URL = '/login';
+
+    /**
+     * @var AcceptanceTester
+     */
+    protected $tester;
+
+    public function __construct(AcceptanceTester $I)
+    {
+        $this->tester = $I;
+    }
+
+    public function login($name, $password)
+    {
+        $I = $this->tester;
+
+        $I->amOnPage(self::$URL);
+        $I->fillField(LoginPage::$usernameField, $name);
+        $I->fillField(LoginPage::$passwordField, $password);
+        $I->click(LoginPage::$loginButton);
+
+        return $this;
+    }    
+}
+?>
+```
+
+And here is an example of how this PageObject can be used in a test.
+
+```php
+<?php
+$I = new AcceptanceTester($scenario);
+$loginPage = new \Page\Login($I);
+$loginPage->login('bill evans', 'debby');
+$I->amOnPage('/profile');
+$I->see('Bill Evans Profile', 'h1');
+?>
+```
+
+If you write your scenario-driven tests in Cest format (which is recommended approach), you can bypass manual creation of PageObject and delegate this task to Codeception. If you specify which objects you need for a test, Codeception will try to create it using it's dependency injection container. In case of PageObject you should declare a class as a parameter for a test method:
```

Comment:
(which is **the** recommended approach)

 bypass manual creation of **a** PageObject

In **the** case of **a** PageObject you should declare a class